### ISSUE TITLE
Tried to cross-sign iOS from Web. both sides claimed it was cancelled despite succeeding.

### ISSUE THREAD

**Issue by ara4n** [vector-im/riot-web#13514]

**Comment by ara4n** [comment]
This leaves me in the state where Web thinks that iOS is verified, but iOS doesn't think that it's verified or cross-sign enabled, which leaves me stuck as i can't unverify iOS from Web. (#13505)

**Comment by manuroe** [comment]
Here is what I found in the log:
* from riot-web:
````
2020-05-05T17:56:02.872Z I PUT /sendToDevice/m.key.verification.cancel/m1588701362872.19 {"@matthew:matrix.org":["STVYVYAPSU"]}
````
* from riot-ios:
````
2020-05-05 18:56:08.027 Riot[2128:632292] [MXCrypto] onToDeviceEvent: event.type: m.key.verification.cancel
2020-05-05 18:56:08.027 Riot[2128:646048] [MXKeyVerification] handleKeyVerificationEvent(from my user: YES, isToDeviceEvent: YES, intendedForMyDevice: MAYBE): eventType: m.key.verification.cancel 
{
    content =     {
        code = "m.unknown";
        reason = "ConnectionError: request failed: CORS request rejected: https://matrix-client.matrix.org/_matrix/client/unstable/keys/signatures/upload";
        "transaction_id" = AEoyFpAmC22Yui3CWujiyQiAYbtfeaU6;
    };
    "origin_server_ts" = 0;
    sender = "@matthew:matrix.org";
    type = "m.key.verification.cancel";
}
```

**Comment by manuroe** [comment]
from riot-web, there is no context that explains why it sent a cancel but it looks like a network issue is the root of the issue.

**Comment by bwindels** [comment]
Yeah, it cancelled because it could not upload the signature. We might want to look into rolling back the verification status if this happens, but I think you should be able to retry from the iOS side as a workaround.

**Comment by ara4n** [comment]
So I just did this again (given matrix.org is now healthy again) - and it completed! but took *ages* (2 minutes?) for the final "did you see a shield?" prompt to respond to OK.  (Plus it looks like the red button has the wrong colour on the CSS - surely this shouldn't be pink)?

<img width="776" alt="Screenshot 2020-05-05 at 18 54 34" src="https://user-images.githubusercontent.com/1294269/81181070-cd98b600-8fa3-11ea-9953-56eea5523126.png">

Have rageshaked again so we can chase down which req took aaaaaaaages (at the least we should retry it, or not block, or make it async, or speed up synapse or something)

**Comment by thinkmassive** [comment]
I experienced a similar issue today. Initiated verification from riot-web and verified the emoji sets, then riot-web showed both devices as verified. However the iOS app showed mobile: red shield, desktop: black shield, "cross-signing is enabled" but "complete security" is still there.

After signing out of the iOS app and signing in again, I was prompted to complete verification via QR code, then both devices show as verified on both ends.
